src/pin.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 8] */
+
+'use strict'
+
+const expect = require('chai').expect
+const fs = require('fs')
+const path = require('path')
+
+const testfile = fs.readFileSync(path.join(__dirname, './data/testfile.txt'))
+
+module.exports = (common) => {
+  describe('.pin', () => {
+    let ipfs
+
+    before(function (done) {
+      // CI is slow
+      this.timeout(20 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist
+          ipfs = node
+          done()
+        })
+      })
+    })
+
+    after((done) => {
+      common.teardown(done)
+    })
+
+    it('add file for testing', (done) => {
+      const expectedMultihash = 'Qma4hjFTnCasJ8PVp3mZbZK5g2vGDT4LByLJ7m8ciyRFZP'
+
+      ipfs.files.add(testfile, (err, res) => {
+        expect(err).to.not.exist
+
+        expect(res).to.have.length(1)
+        expect(res[0].hash).to.equal(expectedMultihash)
+        expect(res[0].path).to.equal(expectedMultihash)
+        done()
+      })
+    })
+
+    describe('callback API', () => {
+      it('.remove (first, because adding is recursive)', (done) => {
```

Comment:
What I meant here was " since ipfs.files.add automatically pins the files, in order to properly test the remaining pin tests, we have to unpin first)"